modules/styling-transform/lib/utils/handleFocusRing.ts

```diff
@@ -0,0 +1,130 @@
+import ts from 'typescript';
+
+import {base, brand} from '@workday/canvas-tokens-web';
+
+import {NodeTransformer} from './types';
+import {parseNodeToStaticValue} from './parseNodeToStaticValue';
+
+export const handleFocusRing: NodeTransformer = (node, checker, prefix, vars) => {
+  // { ...focusRing() }
+  /**
+   * A spread assignment looks like:
+   *
+   * ```ts
+   * {
+   *   ...styles
+   * }
+   * ```
+   *
+   * https://ts-ast-viewer.com/#code/MYewdgzgLgBFCmBbADjAvDA3gMxCAXDAOQBGAhgE5EC+AUKJLAigEzpYB0Xzy1QA
+   */
+  if (ts.isSpreadAssignment(node)) {
+    // Detect `focusRing` calls. This is temporary until we figure out a better way to do focus
+    // rings that doesn't require a special entry in the transform function.
+    //
+    // TODO: implement a fully working type resolver for CSS variables or remove support for them an
+    // remove all uses of `focusRing` from new styling code
+    if (
+      ts.isCallExpression(node.expression) &&
+      ts.isIdentifier(node.expression.expression) &&
+      node.expression.expression.text === 'focusRing'
+    ) {
+      const argumentObject = node.expression.arguments[0];
+      // defaults
+      const defaults = {
+        width: ts.factory.createStringLiteral('2px') as ts.Expression,
+        separation: ts.factory.createStringLiteral('0px') as ts.Expression,
+        inset: ts.factory.createIdentifier('undefined') as ts.Expression,
+        innerColor: ts.factory.createStringLiteral(base.frenchVanilla100) as ts.Expression,
+        outerColor: ts.factory.createStringLiteral(brand.common.focusOutline) as ts.Expression,
+      };
+      if (argumentObject && ts.isObjectLiteralExpression(argumentObject)) {
+        argumentObject.properties.forEach(property => {
+          if (ts.isPropertyAssignment(property) && ts.isIdentifier(property.name)) {
+            defaults[property.name.text as keyof typeof defaults] = property.initializer;
+          }
+        });
+      }
+
+      const inset = parseNodeToStaticValue(defaults.inset, checker, prefix, vars);
+
+      let boxShadow: ts.TemplateExpression;
+      switch (inset) {
+        case 'outer':
```

Comment:
Would `outer` be considered to be both focus rings on the "outside" of the container? I noticed an `inset` here or this something we need to talk about when re-factoring `focusRing`?

Reply:
This is why I think `focusRing` will need an update. `outer` is actually both inner:

https://github.com/Workday/canvas-kit/blob/8817a0a5d258a0f1302ed563beeaa70b40883e5e/modules/react/common/lib/styles/focusRing.ts#L10-L13

I'm supporting `focusRing` in the static optimizer, not changing it.